tgui/packages/tgui/components/MenuBarDropdown.tsx

```diff
@@ -0,0 +1,208 @@
+/**
+ * @file
+ * @copyright 2020 Aleksej Komarov
+ * @license MIT
+ */
+
+import { classes } from 'common/react';
+import { Component, createRef, InfernoNode, InfernoNodeArray, RefObject } from 'inferno';
+import { Box } from './Box';
+import { logger } from '../logging';
+import { Icon } from './Icon';
+
+type MenuProps = {
+  open: boolean;
+  options: InfernoNodeArray;
```

Comment:
Instead of an property that is of type "array", make it "children" instead with "any" type. This makes it simpler to use - you just put options inside the tag, and children can be then directly put where they need to be without checking its length or iterating.

When using an array, you're rendering it inefficiently at best (each item in an array requires a unique key), at worst it can result in hard to debug visual bugs or even bluescreens if elements change/move at any time.

https://reactjs.org/docs/lists-and-keys.html

Reply:
Halfway there on this bit of feedback.